packages/p2p/src/components/my-profile/payment-methods/add-payment-method/select-payment-method.jsx

```diff
@@ -31,7 +31,7 @@ const SelectPaymentMethod = () => {
                                 label={localize('Payment method')}
                                 list_items={my_profile_store.payment_methods_list_items}
                                 onItemSelection={({ value }) => {
-                                    my_profile_store.setSelectedPaymentMethod(value);
+                                    setTimeout(() => my_profile_store.setSelectedPaymentMethod(value), 0);
```

Comment:
There is an issue with the `autocomplete` component used inside a modal, where once a dropdown item is clicked, the `Modal` component detects it as clicking outside the modal.
After taking some time to investigate this issue, it seems that this is occuring due to the switch between `SelectPaymentMethod` and `AddPaymentMethodForm` component (where this switch is conditionally rendered using the `my_profile_store.selected_payment_method` observable changing from the above action) causing the `AddPaymentMethodModal` to not be visible in the screen. During this moment, `setSelectedPaymentMethod` is triggered first, causing the shift in component, and `useOnClickOutside` is triggered **last** and detects this as clicking outside, although its just the switching between two components.
Adding `setTimeout` remedies this by ensuring the priority of this action `my_profile_store.setSelectedPaymentMethod` is called last, allowing `useOnClickOutside` to be invoked first before this action is called and causing the switch between `SelectPaymentMethod` and `AddPaymentMethodForm`